src/App.js

```diff
@@ -14,11 +15,83 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const goods = goodsFromServer.map((item, i) => (
+  {
+    name: item,
+    id: i,
+  }
+));
+
+class App extends React.Component {
+  state = {
+    selectedNames: [],
+    selectedId: [],
+  }
+
+  toggleSelection = (good) => {
+    const state = { ...this.state };
+
+    if (this.state.selectedId.includes(good.id)) {
```

Comment:
you can just check if an array of goods includes a good, an array with ids here seems to be redundant